kqueen/engines/openstack_kubespray.py

```diff
@@ -561,38 +568,60 @@ def __init__(self, stack_name, *, os_kwargs, cluster, extra_ssh_key):
         self.extra_ssh_key = extra_ssh_key
         self.stack_name = stack_name
         self.os_kwargs = os_kwargs
+        self.meta = {}
+
+    def validate_cluster_metadata(self):
+        def validate_ip(address):
+            address = address.strip()
+            # Raises ValueError if IP address is not valid
+            ipaddress.ip_address(address)
+            return address
+
+        mc = self.cluster.metadata["master_count"]
+        if mc % 2 == 0 or mc == 1:
+            raise ValueError("Master node count must be an odd number at least 3 or greater")
+        self.meta["master_count"] = mc
+        self.meta["slave_count"] = self.cluster.metadata["slave_count"]
+
+        self.meta["dns"] = [validate_ip(ip) for ip in
```

Comment:
I suggest to place it after all `is None` checks.

Reply:
why? I placed it here since it performs faster then checks below, so if error will be here, response will be sent faster

Reply:
Why is it faster, I don't get it?

Reply:
since it doesn't send api call to the openstack

Reply:
Oh, okay